pkg/kube/util/boshdns/interface.go

```diff
@@ -22,14 +22,16 @@ type DomainNameService interface {
 // New returns the DNS service management struct
 func New(m bdm.Manifest) (DomainNameService, error) {
 	dns := NewBoshDomainNameService(m.InstanceGroups)
-
-	index := HasBoshDNSAddOn(m)
-	if index != -1 {
-		if err := dns.Add(m.AddOns[index]); err != nil {
-			return nil, errors.Wrapf(err, "error loading BOSH DNS configuration")
+	for index, addon := range m.AddOns {
+		for _, job := range addon.Jobs {
+			if job.Release == bdm.BoshDNSAddOnName {
```

Comment:
This needs to look at 'bosh-dns' and 'bosh-dns-aliases' I think.

Reply:
Done.